tests/llama_index/test_llama_index_model_export.py

```diff
@@ -0,0 +1,259 @@
+
+    loaded_model = mlflow.llama_index.load_model(logged_model.model_uri)
+
+    assert "llama_index" in logged_model.flavors
+    assert type(loaded_model) == type(index)
+    engine = loaded_model.as_query_engine()
+    assert engine is not None
+    assert engine.query("Spell llamaindex").response != ""
+
+
+@pytest.mark.parametrize(
+    "engine_type",
+    ["query", "retriever"],
+)
+def test_format_predict_input_correct(single_index, engine_type):
+    wrapped_model = create_engine_wrapper(single_index, engine_type)
+
+    assert isinstance(
+        wrapped_model._format_predict_input(pd.DataFrame({"query_str": ["hi"]})), QueryBundle
+    )
+    assert isinstance(wrapped_model._format_predict_input(np.array(["hi"])), QueryBundle)
+    assert isinstance(wrapped_model._format_predict_input({"query_str": ["hi"]}), QueryBundle)
+    assert isinstance(wrapped_model._format_predict_input({"query_str": "hi"}), QueryBundle)
+    assert isinstance(wrapped_model._format_predict_input(["hi"]), QueryBundle)
+    assert isinstance(wrapped_model._format_predict_input("hi"), QueryBundle)
+
+
+@pytest.mark.parametrize(
+    "engine_type",
+    ["query", "retriever"],
+)
+def test_format_predict_input_incorrect_schema(single_index, engine_type):
+    wrapped_model = create_engine_wrapper(single_index, engine_type)
+
+    with pytest.raises(ValueError, match="correct schema"):
+        wrapped_model._format_predict_input(pd.DataFrame({"incorrect": ["hi"]}))
+    with pytest.raises(ValueError, match="correct schema"):
+        wrapped_model._format_predict_input({"incorrect": ["hi"]})
+
+
+@pytest.mark.parametrize(
+    "engine_type",
+    ["query", "retriever"],
+)
+def test_format_predict_input_correct_schema_complex(single_index, engine_type):
+    wrapped_model = create_engine_wrapper(single_index, engine_type)
+
+    payload = {
+        "query_str": "hi",
+        "image_path": "some/path",
+        "custom_embedding_strs": [["a"]],
+        "embedding": [[1.0]],
+    }
+    assert isinstance(wrapped_model._format_predict_input(pd.DataFrame(payload)), QueryBundle)
+    payload.update(
+        {
+            "custom_embedding_strs": ["a"],
+            "embedding": [1.0],
+        }
+    )
+    assert isinstance(wrapped_model._format_predict_input(payload), QueryBundle)
+
+
+@pytest.mark.parametrize("with_signature", [True, False])
+def test_query_engine_str(tmp_path, single_index, with_signature):
+    payload = "string"
+
+    signature = infer_signature(model_input=payload) if with_signature else None
+    mlflow.llama_index.save_model(
+        index=single_index, signature=signature, path=tmp_path, engine_type="query"
+    )
+    model = mlflow.pyfunc.load_model(tmp_path)
+    predictions = model.predict(payload)
+    assert isinstance(predictions, Response)
+    assert predictions.response
+
+
+@pytest.mark.parametrize("with_signature", [True, False])
+def test_query_engine_numeric(tmp_path, single_index, with_signature):
```

Comment:
Can we also use `@pytest.mark.parametrize` for these different input data types to reduce boilerplate code?

Reply:
Given most have different return asserts, I can only condense 4 functions down to 2. For example `test_chat_engine_dict` and `test_chat_engine_str`. 
1. Should I condense these functions or leave as is?
2. Do you think it's a good idea to rewrite the asserts to be more generic and sacrifice coverage quality? We could probably reduce test count by ~8 with that.